### DISCUSSION TITLE
Packet timestamping: How can I get the timestamps with libssl?

### DISCUSSION THREAD

**Discussion by kutoga** [openssl/openssl#22131] [Q&A] [answered]
Hello

With Linux it's possible to timestamp incoming network data either on the network card or in the kernel (https://www.kernel.org/doc/Documentation/networking/timestamping.txt).

Here's an example code to do this: https://raw.githubusercontent.com/majek/openonload/master/src/tests/onload/hwtimestamping/rx_timestamping.c

(The code depends on using the `recvmsg` function / syscall).

Is it somehow possible to get these timestamps when using libssl? So far I use `SSL_read`, but it seems this finally uses `read` (and not `recvmsg`). If it's not possible, does someone has a hint what part of the code I would need to modify in libssl in order to get that working?

Thanks a lot &

Best regards,
kutoga

**Comment by mattcaswell** [comment, marked as answer]
Co-incidentally I just answered your stack overflow question on this:

https://stackoverflow.com/a/77125987/1946679

**Comment by kutoga** [comment]
Yep, I just saw that. Thanks :smile:
